File: scripts/sync_seed.js
import fs from 'fs';
import path from 'path';

const tsPath = 'c:/Users/User/Downloads/PortableGit/Learn Opportunities/src/data/opportunities.ts';
const seedPath = 'c:/Users/User/Downloads/PortableGit/Learn Opportunities/backend/seed.js';

let tsContent = fs.readFileSync(tsPath, 'utf8');

// Pull the array content
const match = tsContent.match(/export const opportunities: Opportunity\[\] = \[\s*([\s\S]*)\s*\];\s*$/);
if (!match) {
    console.error("No opportunities found!");
    process.exit(1);
}

let arrayContent = match[1];

// Add views, clicks to each item
// We'll replace the closing brace of each item with our extra properties
// A better way is to regex for `},` and add it before the comma
let updatedArray = arrayContent.replace(/\},/g, '    views: 0,\n    clicks: 0\n  },');

// Add to the final item (since it doesn't have a trailing comma after the array content match)
// but wait, the last item might have a comma if the dev left it.
// Checking the actual content from earlier: `logoUrl: '/images/opportunities/google.jpg'\n    }\n];`
// So it doesn't have a comma after the last one.
updatedArray = updatedArray.trim();
if (!updatedArray.endsWith('},')) {
    // replace the last } with views/clicks + }
    updatedArray = updatedArray.replace(/\}\s*$/, '    views: 0,\n    clicks: 0\n  }');
}

let seedContent = fs.readFileSync(seedPath, 'utf8');

const newSeed = seedContent.replace(/const opportunities = \[\s*[\s\S]*?\s*\];/ , `const opportunities = [\n${updatedArray}\n];`);

fs.writeFileSync(seedPath, newSeed);
console.log("Seed file successfully updated using node!");

File: scripts/sync_seed.py
import re
import json

def transform_ts_to_js(ts_content):
    # Find the opportunities array content
    match = re.search(r'export const opportunities: Opportunity\[\] = \[(.*)\];', ts_content, re.DOTALL)
    if not match:
        print("Opportunities list not found!")
        return None
    
    array_content = match.group(1).strip()
    
    # We'll parse this later but for now we'll do some basic string manipulation
    # To add "views: 0, clicks: 0" inside every opportunity object
    
    # Each object ends with a "}" followed by optional whitespace and a comma
    # We'll replace it with "  views: 0,\n    clicks: 0\n  },"
    
    transformed = re.sub(r'\},', r'    views: 0,\n    clicks: 0\n  },', array_content)
    # Also handle the last object
    if not transformed.strip().endswith('},'):
         transformed = transformed.strip() + ',\n    views: 0,\n    clicks: 0\n  '
    
    return transformed

with open("c:/Users/User/Downloads/PortableGit/Learn Opportunities/src/data/opportunities.ts", "r", encoding="utf-8") as f:
    ts = f.read()

js_array = transform_ts_to_js(ts)

# Now read the current seed.js
with open("c:/Users/User/Downloads/PortableGit/Learn Opportunities/backend/seed.js", "r", encoding="utf-8") as f:
    seed0 = f.read()

# Replace "const opportunities = [ ... ];" with new list
# Use re.DOTALL to match multiline
new_seed = re.sub(r'const opportunities = \[.*?\];', f'const opportunities = [\n{js_array}\n];', seed0, flags=re.DOTALL)

with open("c:/Users/User/Downloads/PortableGit/Learn Opportunities/backend/seed.js", "w", encoding="utf-8") as f:
    f.write(new_seed)

print("Seed file successfully updated!")
